src/include/misc.h

```diff
@@ -34,6 +34,9 @@
  * return values.
  */
 #define WT_UNUSED(var) (void)(var)
+
+/* Quiet clang static analyzer warnings about dead stores. */
+#define WT_DEAD_STORE(var) WT_UNUSED(var)
```

Comment:
This is inline with the documentation on how to deal with false positives, technically gcc is the one with the error but suppressing gcc is much uglier.

Docs: https://clang-analyzer.llvm.org/faq.html